tests/NotificationTests/NotificationCenterTest.php

```diff
@@ -0,0 +1,639 @@
+        );
+        $this->notificationCenterObj->addNotificationListener(
+            NotificationType::TRACK,
+            array($clientMock, 'track_callback_no_args')
+        );
+
+        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        // === Verify that all callbacks for NotificationType::DECISION are called and no other callbacks are called === //
+        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+        $clientMock->expects($this->exactly(1))
+            ->method('decision_callback_no_args');
+        $clientMock->expects($this->exactly(1))
+            ->method('decision_callback_no_args_2');
+
+        $clientMock->expects($this->never())
+            ->method('track_callback_no_args');
+
+        $this->notificationCenterObj->fireNotifications(NotificationType::DECISION);
+    }
+
+    public function testFireNotificationsAndVerifyThatAllCallbacksWithArgsAreCalled()
+    {
+        $clientMock = $this->getMockBuilder(FireNotificationTester::class)
+            ->setMethods(array('decision_callback_with_args', 'decision_callback_with_args_2', 'track_callback_no_args'))
+            ->getMock();
+
+        $this->notificationCenterObj->clearAllNotifications();
+
+        //add notification callbacks with args
+        $this->notificationCenterObj->addNotificationListener(
+            NotificationType::DECISION,
+            array($clientMock, 'decision_callback_with_args')
+        );
+        $this->notificationCenterObj->addNotificationListener(
+            NotificationType::DECISION,
+            array($clientMock, 'decision_callback_with_args_2')
+        );
+        $this->notificationCenterObj->addNotificationListener(
+            NotificationType::TRACK,
+            array($clientMock, 'track_callback_no_args')
+        );
+
+        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        // === Verify that all callbacks for NotificationType::DECISION are called and no other callbacks are called === //
+        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+        $clientMock->expects($this->exactly(1))
+            ->method('decision_callback_with_args')
+            ->with(5, 5.5, 'string', array(5,6), function () {
+            });
+        $clientMock->expects($this->exactly(1))
+             ->method('decision_callback_with_args_2')
+             ->with(5, 5.5, 'string', array(5,6), function () {
+             });
+        $clientMock->expects($this->never())
+            ->method('track_callback_no_args');
+
+        $this->notificationCenterObj->fireNotifications(
+            NotificationType::DECISION,
+            array(5, 5.5, 'string', array(5,6), function () {
+            })
+        );
+    }
+}
```

Comment:
add empty line :P 
btw, there are some more test cases associated with activate, track & is_featureenabled to verify that firenotification is called or not. have u done those/

Reply:
I will do them today in OptimizelyTest.php